src/main/scala/com/amazon/deequ/analyzers/catalyst/AttributeReferenceCreation.java

```diff
@@ -37,6 +39,16 @@ private AttributeReferenceCreation() { }
         signature in different versions of Spark 2.x. Therefore we need to invoke it via reflection
         depending on which version of Spark we run.
 
+        SPARK 2.4:
```

Comment:
We could add more safe guards around the if checks. For example any `AttributeReference` with 7 parameters is invoked with Spark 2.2 signature. Can we rather check all of the arguments of `AttributeReference` instead?

Reply:
There are a lot of things we can check for sure. My preference is to "let it crash" in the tests, rather than resorting to defensive programming. Maybe you can elaborate more on your view: what would you check, and - more importantly - how exactly would you deal with previously unseen signatures?

Reply:
I dont think we have to handle unseen signatures, we should just make sure to correctly handle all major Spark 2.x versions

Reply:
I meant to check the whole signature of all known versions instead of checking the number of arguments only.

Reply:
That's what I understood. Personally, I don't think that the defensive approach adds much here because 1/ we discover problems when running the test suite for a new Spark release, 2/ when discovering an unexpected argument type we are already in an illegal state and there's no remedy, so the only benefit we gain is a more informative exception. With these points combined, I'm not in favor of checking the signature exhaustively.

Reply:
Good point, fully agree. A more informative exception can help in pinpointing the issue for new Spark releases.